extensions/emmet/src/util.ts

```diff
@@ -128,6 +128,97 @@ export function parseDocument(document: vscode.TextDocument, showError: boolean
 	return undefined;
 }
 
+export function parsePartialStylesheet(document: vscode.TextDocument, position: vscode.Position): Node | undefined {
+
+	let startPosition = new vscode.Position(0, 0);
+	let endPosition = new vscode.Position(document.lineCount - 1, document.lineAt(document.lineCount - 1).text.length);
+	const closeBrace = 125;
+	const openBrace = 123;
+	let slash = 47;
+	let star = 42;
+
+	// Go forward until we found a closing brace.
+	let stream = new DocumentStreamReader(document, position);
+	while (!stream.eof()) {
+		if (stream.eat(closeBrace)) {
+			break;
+		} else if (stream.eat(slash)) {
+			if (stream.eat(slash) && document.languageId !== 'css') {
+				// Single line Comment, we continue searching from next line.
+				stream.pos = new vscode.Position(stream.pos.line + 1, 0);
+			} else if (stream.eat(star)) {
+				// Start of block comment, we need to find the closing '*/'
+				let endCommentFound = false;
+				while (!endCommentFound) {
```

Comment:
This will get stuck in an infinite loop if `*/` was not found and you reached eof